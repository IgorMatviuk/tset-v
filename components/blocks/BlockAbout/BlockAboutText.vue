<template>
  <div class="about-text">
    <div class="about-text__title" v-html="content.title"></div>
    <div class="about-text__content" v-html="content.description"></div>
  </div>
</template>

<script>
export default {
  props: {
    content: {
      type: Object,
      required: true
    }
  }
}
</script>

<style lang="scss">
.about-text {
  padding-left: 15px;

  &__title {
    position: relative;
    margin-bottom: 31px;
    font-size: 16px;
    line-height: 20px;
    text-transform: uppercase;
    letter-spacing: 0.05em;

    &::before {
      position: absolute;
      bottom: -16px;
      left: -15px;
      width: 90px;
      height: 1px;
      content: '';
      background-color: $primary-light-1;
    }
  }

  &__content {
    padding-right: 5px;
    font-size: 12px;
    line-height: 15px;
    color: $text-color;

    p {
      margin-bottom: 14.5px;
      font-size: 12px;
      line-height: 15px;
      color: $text-color;
    }
  }

  @include media-breakpoint-up(md) {
    &__title {
      margin-bottom: 36px;

      &::before {
        bottom: -21px;
        left: -20px;
        width: 110px;
      }
    }
  }

  @include media-breakpoint-up(xl) {
    &__title {
      margin-bottom: 30px;
      font-size: 18px;
      line-height: 23px;

      &::before {
        bottom: -15px;
        left: -41px;
      }
    }

    &__content {
      font-size: 13px;
      line-height: 16px;

      p {
        margin-bottom: 16px;
        font-size: 13px;
        line-height: 16px;
      }
    }
  }
}
</style>
