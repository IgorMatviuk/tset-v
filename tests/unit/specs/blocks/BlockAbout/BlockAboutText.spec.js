import { shallowMount } from '@vue/test-utils'

import BlockAboutText from '@/components/blocks/BlockAbout/BlockAboutText'

describe('BlockAboutText', () => {
  let propsData
  let content

  beforeEach(() => {
    content = {
      title: `<h1>BlockAboutText content</h1>`,
      description: `<p>BlockAboutText description</p>`
    }

    propsData = {
      content
    }
  })

  test('has expected structure', () => {
    const wrapper = shallowMount(BlockAboutText, {
      propsData,
      mocks: {
        $t: () => ''
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  test('rendering content', () => {
    const wrapper = shallowMount(BlockAboutText, {
      propsData,
      mocks: {
        $t: () => ''
      }
    })

    const title = wrapper.find('.about-text__title > *')
    const description = wrapper.find('.about-text__content > *')

    expect(title.html()).toBe(content.title)
    expect(description.html()).toBe(content.description)
  })
})
