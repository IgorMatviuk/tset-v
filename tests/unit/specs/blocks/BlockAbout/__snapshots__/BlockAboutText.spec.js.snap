// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockAboutText has expected structure 1`] = `
<div
  class="about-text"
>
  <div
    class="about-text__title"
  >
    <h1>
      BlockAboutText content
    </h1>
  </div>
   
  <div
    class="about-text__content"
  >
    <p>
      BlockAboutText description
    </p>
  </div>
</div>
`;
